### DISCUSSION TITLE
Current Development

### DISCUSSION THREAD

**Discussion by kn1meR** [kn1meR/CFB-Dynasty-Manager#3] [Announcements]
Here is what is being worked on since the initial v2026 release:

<ins>COMPLETED</ins>

- ~~Update positional terminology~~
- ~~Allow changing of conferences easily.~~
- ~~Fix Top 25 logic.~~
- ~~Add Rankings wherever possible.~~
- ~~Update the UI buttons on the Recruiting & Transfers pages.~~
- ~~Update Player Award names.~~
- ~~Automatically add Recruits / Transfers to Roster upon ending year.~~
- ~~Automatically progress player's Year / Class upon ending year.~~
-  ~~Enhance the Player Card UI.~~

<ins> TO DO </ins>

- Revamp the Social Media logic.
- Revamp Season Results card in Season History.
- Enhance support for TeamBuilder schools.
- Changing schools will retroactively change Season History to new school. (bugfix)

Any suggestions / recommendations are welcome!  Please also report any bugs or issues that you find.

**Comment by McKapy** [comment]
FCS schools need added to the school list. Also, an addition to have a dropdown for awards, rather than manual entry

**Reply by kn1meR** [reply]
WanderingCard has added the FCS Teams to the team list and i just added a dropdown for the Awards page!

**Comment by WanderingCard** [comment]
Not too familiar with radix, and couldn't find one, so I am going to ask if you know how to make one: Would probably be much better to make the team selects be an autocomplete, if one exists in radix, instead of just a select box (I.E type in text, and the select menu filters) Given there is a 132 teams, having them all dumped into one massive select menu is a bit unwieldly, especially for schools in the middle of the alphabet

**Comment by TheBigHouse7** [comment]
This is awesome! A few ideas that came to mind for the roster page:

- Similar to the "potential" dropdown for recruiting, would be great to have the option to list dev traits for existing players in the 2024 roster.
- Was also thinking a "Notes" column that's just a text field for each player would be cool for the roster. Could list things like if a player is a walk-on, captain, that they're from the "Class of [year]", etc.
- Add FB and specific positions (LG, DE, LOLB, SS, etc.) for the offensive line, defensive line, linebackers, and safety.

**Reply by kn1meR** [reply]
Thanks for the suggestions!  I just pushed some updated files that add the above features to the roster page!

**Reply by TheBigHouse7** [reply]
Looks great! After inputting my full roster, had some additional observations/ideas:

- For the newly separate positions (LG, DE, etc.), I wasn't able to save players with them. However, I could add players with different positions and then update them to the correct position (e.g., putting a player as a TE and then updating them to LT).
- In light mode, the text in the upper-right corner isn't visible.
<img width="400" alt="Screenshot 2024-09-16 at 10 10 36 PM" src="https://github.com/user-attachments/assets/e5a16c01-1651-47ab-8be2-4291cc6717d6">

- Idea: When it comes to sorting, I'm not sure what's possible, but was wondering if it'd be feasible to do a custom sort order to match the order of the positions in the dropdown (instead of alphabetical). Along the same lines, when sorting by position, would it be feasible to order the players by rating? Looks like it's currently based on the order in which players were added.
- Idea: Add a hover state for each row or alternating colors for rows to better differentiate between them.
- Idea: Add a # column before the name to list the jersey number.

**Reply by kn1meR** [reply]
Appreciate the ideas!  I believe that WanderingCard pushed a commit to fix the position errors, and I made some updates that change the coach profile, adds hover FX for the roster page, and a sortable jersey # column!

**Comment by PubliusRusticus** [comment]
Is there a way to add a TeamBuilder team to this?

**Reply by WanderingCard** [reply]
If you go into the repository folder, then to src and utils there should be a fbsTeams.ts file.

That contains the information on all the teams the program recognizes, if you just find the object for the team the teambuilder is replacing, and change the values to match the team builder, and save it, that should work.

**Comment by wnorris347** [comment]
Might be good to add All-American/All-Conference to either Player Awards or the Season Stats screen. I use Notes to keep track right now but it tends to get a little crowded if a player makes the list multiple years.

**Reply by kn1meR** [reply]
thanks for the suggestion!  I have added All-American and All-Conference as Awards in the Player Awards for easier tracking!

**Comment by Coobsie** [comment]
Is it possible to have two separate instances of this? For example, one instance for my solo dynasty and an instance for a duo dynasty to keep track of both separately? If so, how? Apologies if that's a silly question, new to github.

**Reply by WanderingCard** [reply]
It's possible but might be a bit complicated. 

The way data is stored uses the browser's internal storage, specific to the "webpage" being rendered, in this case its localhost:3000

If you load up the project in a different browser (I.E have one in chrome, another in Firefox) or possibly a different browser profile, each would have its own data and not interact I believe, but you have to be careful to:

1) remember which browser/profile is which dynasty
2) Never delete website data on either browser or profile, otherwise it will purge all data in the tracker

**Reply by kn1meR** [reply]
If you're using the most recent version of the application, there is an Import / Export functionality that you could utilize.  For one dynasty, just export and name it.  Then clear the data, enter your other dynasty info, export & name it.  Then you should be able to import the desired dynasty whenever you want.  Just remember to export every time you exit or you may lose tracked progress.

**Comment by CFBDuckMania** [comment]
Super excited to use this for 26! Thank you for contributing to an amazing community.

**Reply by CFBDuckMania** [reply]
One tweak is that Villanova is not a team available in the drop down under schedule.

**Reply by kn1meR** [reply]
this only includes the teams in the game.  since Villanova is not a playable team, they are not included.  & hope you enjoy it!  be on the lookout for an update in the coming days ;)

**Comment by n8faz** [comment]
This is sick. Stoked to use this tomorrow. Is there any way you can make it so we can edit a team's conference? A whole conference tab to track realignment and standings would be amazing, but just the ability to select which conference a school is in would be great. Thanks for this program really wish I had it last year!

**Reply by kn1meR** [reply]
you can always edit the file containing all of the team information.  the file is **/src/utils/fbsTeams.ts** .  Just open that file with a text editor, and change the conference of your school to one of the ones at the top of the file (must match 1-1).  i will look into adding a Conference tab!

**Reply by n8faz** [reply]
Editing that file worked perfectly thank you!

**Comment by kn1meR** [comment]
In the latest update, I made a change to the position terminology to reflect the new LEDGE. REDGE, SAM, WILL, and MIKE positions.  If your current roster has any players with the old positions, make sure you edit them and update to the new position!

**Comment by JimJamTC** [comment]
First of all this is awesome thanks so much for the hard work put into this. Just wondering if there was anyway a standings and playoff tree section could be added because i really want to use this but those are like the 2 most important things i like to go back and look at in my dynasties

**Comment by kn1meR** [comment]
Just released an update! As always, please report any bugs / issues!

Version CFBDM-v2605 can be found here:
https://github.com/kn1meR/CFB-Dynasty-Manager/releases/tag/CFBDM-v2605

**Comment by jwbw29** [comment]
I messaged you on reddit but you can ignore it -- i've been scattered. I'm gonna stick to keeping everything on here now that I see there's a history of discussion

Here was my initial finding:

> I noticed you added a change that doesn't display weeks in the top 25 until the week has been advanced. But how do you advance weeks? In my schedule I have 3 games played with results and scores.
>
> But Top 25 is only showing week 0
>
> And on the home page, it shows the most recent result, but not all previous results which makes me think it's only reading current week some how

After some digging I think I found the issue and have applied a fix in my forked repo. The link to the commit is below if you wanna check it out. 

I also updated the "Recent Results" to exclude Bye games because for some reason it was looking ahead. For example, I was on week 5 and expected to see results from Week 0, Week 1, Week 3, and Week 4 (because week 2 was a BYE), but instead I was seeing results from Week 3 and Week 4, and then Week 9, Week 14, and Week 16 showing BYE weeks.

https://github.com/jwbw29/CFB-Dynasty-Manager/commit/b59386f0adb36e1a8c70dd42fdffcf1f82cefec2

**Reply by kn1meR** [reply]
i will take a look at your fix for the Top 25! 

>I noticed you added a change that doesn't display weeks in the top 25 until the week has been advanced. But how do you advance weeks? In my schedule I have 3 games played with results and scores.
>
>But Top 25 is only showing week 0
>
>And on the home page, it shows the most recent result, but not all previous results which makes me think it's only reading current week some how

Recent Results, like Upcoming Games,  will only show the 5 most recent weeks.

This was a bug that was caused by the week being reset to whatever previous week was selected in the dropdown application wide. Once selected, it thought that the user was in that week and would not let them return to the current week.  A fix was implemented for this issue.
 
as for the recent results showing bye weeks in advance, i ran into that issue as well and added a fix for it.  

the way advancing a week works is when a result is entered on the schedule, that acts as "week complete" so it will advance.  the issue was that any time you had a Bye week scheduled, it knew that the result was going to be "Bye" and not W / L / T, so it automatically propogated the result, thus advancing those weeks.  That is not the behavior that was desired, so I removed the auto result completion, making it so the user will enter the "Bye" result once that week has been completed.  I didn't want to exclude Bye weeks on the Recent Results because then the weeks would not be linear like one would expect.

TLDR: results should only be entered for games / weeks that have been completed.  If a BYE result in week 14 has been entered, and you are only in week 4, this may cause unexpected behavior. 

Hopefully that answers your questions!

**Reply by jwbw29** [reply]
ahhhhh ok. So best use then is: if you wanna enter your whole schedule ahead of time you can do that...just don't populate BYE in the result, only in the opponent? That makes sense. I'll give this a go

**Reply by jwbw29** [reply]
yup that was the issue. Thanks for clarifying that!

Just outta curiosity -- is there any intent to let custom conferences be unique to each dynasty? Hoping to start using this for my online dynasties but would need to implement something for the custom conferences set in each one. If not, then I may work on that on the side

**Reply by kn1meR** [reply]
Correct.  You can populate the schedule with opponents and locations, just no results until you complete that week!

**Reply by kn1meR** [reply]
> yup that was the issue. Thanks for clarifying that!
> 
> Just outta curiosity -- is there any intent to let custom conferences be unique to each dynasty? Hoping to start using this for my online dynasties but would need to implement something for the custom conferences set in each one. If not, then I may work on that on the side

that would be an interesting idea to work on.  i'm not sure how that would work and probably not not on the shortlist of things I'd work on implementing, but down the line could look into it.  If you want to take a crack at it, then you are more than welcome!

**Comment by jwbw29** [comment]
Another quick question for ya - sorry to be annoying about all this. What's the recommended way to use this app if you switch teams in a Dynasty? I did some testing and noticed that all of the styling and branding does accurately change. But I also noticed that when you go to Season History and go to a previous year, it still shows the current team even for past seasons.

Is this just a feature that hasn't been implemented yet, or is the intended use to create a new dynasty if you switch teams (even w/in the same dynasty on the game)?

**Reply by kn1meR** [reply]
no you're good!  i was made aware of this and have it at the top of my to-do list.  ideally, you should be able to just switch teams and the past seasons should retain your old school's logos and info, while your current season should retain the new school.  starting a new dynasty is a valid option, but then it feels like its a new coach when it is not.

**Reply by jwbw29** [reply]
Sick. This thing is quickly becoming my main tracker and am slowly abandoning my Sheets. I think you're super close to being able to start making some money on this thing lol

**Reply by kn1meR** [reply]
haha thanks!  i'm glad you're enjoying it!

i don't have any plans on charging for this as i want to keep it accessible to everyone.  open source and free is the way to be!

**Comment by kn1meR** [comment]
Just released an update addressing a critical issue! As always, please report any bugs / issues!

The updated version `CFBDM-v2606` can be found here:

https://github.com/kn1meR/CFB-Dynasty-Manager/releases/tag/CFBDM-v2606

**Comment by jackwardphillips** [comment]
I've been putting off making something like this forever, this is pretty much exactly what I was looking for. I think a good idea to the roster screen would be including dealbreakers. I think the transfer logic could be hardcoded, I'm not sure how much it varies. With how crazy the transfer portal is this year could be a valuable addition. I'm also a sap for storylines so a hometown column would be awesome too, but I can also throw that in notes.

**Comment by Johnyboylmao** [comment]
Ive never seen anything like this and this is awesome bro thanks for putting time into it and making it free!!

**Comment by kn1meR** [comment]
Just released an update adding some features and making some changes! As always, please report any bugs / issues!

The updated version `CFBDM-v2608` can be found here:

https://github.com/kn1meR/CFB-Dynasty-Manager/releases/tag/CFBDM-v2608
